### PR TITLE
test: add testing for requests.go

### PR DESCRIPTION
**Pull request by dvirgilad**

### CODE REVIEW

File: internal/issuer/clients/cert/requests_test.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	PostCert         = "PostCert"
+	GetCert          = "GetCert"
+	FailGetCert      = "FailGetCert"
+	InvalidResponse  = "InvalidResponse"
+	TestURL          = "https://test.com/"
+	DownloadEndpoint = "download/"
+	GUID             = "12345678/"
+)
+
+func TestRequests(t *testing.T) {
```
Comment (mzeevi) [inline internal/issuer/clients/cert/requests_test.go:27]: Each function should be tested separately, so there should be a `TestPostCertificate` and `TestDownloadCertificate`

File: internal/issuer/clients/cert/requests_test.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	type want struct {
+		method    string
+		path      string
+		responder httpmock.Responder
+	}
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"ShouldSendPostRequest": {
```
Comment (mzeevi) [inline internal/issuer/clients/cert/requests_test.go:45]: `ShouldSendPOSTRequestSuccessfully`

File: internal/issuer/clients/cert/requests_test.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+				client: NewClient(WithAPIEndpoint(TestURL), WithHTTPClient(hClient)),
+			},
+			want: want{
+				method: http.MethodPost,
+				path:   TestURL + "csr",
+				responder: func(request *http.Request) (*http.Response, error) {
+					requestBody, _ := io.ReadAll(request.Body)
+					if !bytes.Equal(requestBody, exampleBytes) {
+						return httpmock.NewJsonResponse(http.StatusBadRequest, fmt.Sprintf("got %b, want %b", request.Body, exampleBytes))
+					}
```
Comment (mzeevi) [inline internal/issuer/clients/cert/requests_test.go:57]: Why is this needed?

File: internal/issuer/clients/cert/requests_test.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+var exampleBytes = []byte("example")
+
+const (
+	PostCert         = "PostCert"
+	GetCert          = "GetCert"
+	FailGetCert      = "FailGetCert"
+	InvalidResponse  = "InvalidResponse"
+	TestURL          = "https://test.com/"
+	DownloadEndpoint = "download/"
+	GUID             = "12345678/"
```
Comment (mzeevi) [inline internal/issuer/clients/cert/requests_test.go:24]: Why are these constants exported?

File: internal/issuer/clients/cert/requests_test.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+				responder: func(request *http.Request) (*http.Response, error) {
+					requestBody, _ := io.ReadAll(request.Body)
+					if !bytes.Equal(requestBody, exampleBytes) {
+						return httpmock.NewJsonResponse(http.StatusBadRequest, fmt.Sprintf("got %b, want %b", request.Body, exampleBytes))
+					}
+					return httpmock.NewJsonResponse(http.StatusOK, nil)
+				},
+			},
+		},
+		"ShouldSendGetRequest": {
```
Comment (mzeevi) [inline internal/issuer/clients/cert/requests_test.go:62]: `ShouldSendGETRequestSuccessfully`

File: internal/issuer/clients/cert/requests_test.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+				client: NewClient(
+					WithAPIEndpoint(TestURL),
+					WithForm(testForm),
+					WithDownloadEndpoint(DownloadEndpoint),
+					WithHTTPClient(hClient),
+				),
+			},
+			want: want{
+				method: http.MethodGet,
+				path:   TestURL + GUID + DownloadEndpoint + testForm,
```
Comment (mzeevi) [inline internal/issuer/clients/cert/requests_test.go:74]: Maybe form the path with a `fmt.Sprintf()` for better readability?

**Review by mzeevi — CHANGES_REQUESTED** [review]

File: internal/issuer/clients/cert/requests_test.go (hunk 3 above)
Comment (dvirgilad) [reply internal/issuer/clients/cert/requests_test.go:57]: testing to see it sends the certificate correctly

**Review by mzeevi — APPROVED** [review]
/lgtm
/approve

**Review by mzeevi — APPROVED** [review]
/lgtm
/approve
